python_benchmarks/global_params.py

```diff
@@ -0,0 +1,27 @@
+import pytest
+import torch
+
+# Correctness Tests tolerance
+RTOL = 1e-3
+ATOL = 1e-3
+
+
+# Common parameters for all tests
+def generate_input_sizes():
+    range_outer = [2**i for i in range(1, 5)]
+    range_inner = [32 * 1024 * 2**i for i in range(11)]
+
+    inputs = [(i, j) for i in range_outer for j in range_inner]
+    inputs.extend([(j, i) for i in range_outer for j in range_inner])
+    return inputs
+
+
+INPUT_SIZES = generate_input_sizes()
+DTYPES = [torch.float16, torch.float32]
+TEST_CORRECTNESS = [True]
+
+pytestmark = [
+    pytest.mark.parametrize("size", INPUT_SIZES),
+    pytest.mark.parametrize("dtype", DTYPES),
+    pytest.mark.parametrize("test_correctness", TEST_CORRECTNESS),
+]
```

Comment:
This seems to be unused. I would think explicitly parametrizing each benchmark will be preferable. The input sizes and dtypes that we are interested in will change in each benchmark probably.

Reply:
They are implicitly used in the files where it is imported. 
Since a large number of benchmarks, do use this set of inputs and datatypes, it does simplify the benchmarks and avoids the parametrize calls for those. For benchmarks that deviate from this, I would avoid importing this and setting inputs and dtypes directly.

Reply:
But it's imported at the module level right? Can we override these parametrizations for a single test within a module? The @parametrize line is explicit and not too verbose so I'd personally prefer to have them local to the test. Casually reading through the benchmarks it might be unclear where to find or alter them otherwise.

Reply:
Right, it is at the module level and can be a bit of a blackbox.
In that case, I will get rid of pytestmark completely. 

@naoyam What is your view?

I am also removing the flag for `test_correctness` since any module that imports pytestmark will need to pass this argument irrespective of whether it is required. The current benchmarks will also validate the results but given the benchmarks will run the function for several iterations, it should not be a significant overhead.

Reply:
I think what's important here is that if there's a common benchmark parameter, we should make it automatically included in all benchmarks in a consistent way. That would make sure the parameter is used in a consistent way across benchmarks, which would be important for post-benchmarking automated analyses.

As @jacobhinkle mentioned, however, each benchmark will have different parameter values, so we need some way to define benchmark-specific parameter values.

Reply:
> I am also removing the flag for test_correctness since any module that imports pytestmark will need to pass this argument irrespective of whether it is required

Can you make it possible to just run the correctness run only? That'd be also a common usage of benchmarks.

Reply:
I think this section of the docs shows a possible way for us to add --disable-correctness-checks and --disable-benchmarking options, which would let us run in pure benchmark or pure test modes without using parametrize which clutters the test name: https://docs.pytest.org/en/7.1.x/example/simple.html#pass-different-values-to-a-test-function-depending-on-command-line-options.

Reply:
Thanks @jacobhinkle. Added `--disable-validation` and `--disable-benchmarking` as command line options.